src/GameLogic.cpp

```diff
@@ -337,17 +368,27 @@ void GameLogic::updateEnemyMovement(Enemy& enemy, float timeS) {
         int dir = rand() % 2;
         float currVelX = ( ((dir == 0) ? (-1*enemy.getStepSize()) : enemy.getStepSize()) * timeS );
         enemy.setVelocityX(currVelX);
+		enemy.setDirection(dir);
     }
     enemyFall(enemy, timeS);
     enemyTrack(enemy, timeS);
-    enemy.updateMovement(timeS);
+    enemy.updateMovement();
 }
 
-void GameLogic::playerShoot(float timeS){
-    if ( ! playerFired ) {
-        playerFired = true;
-        projectile.init(1,player.getSprite().getPosition().x+20,player.getSprite().getPosition().y + 15, player.getDirection());
-        //projectiles.push_back(projectile);
-        //actorsVector.push_back(projectile);
+void GameLogic::playerShoot(){
+    if ( player.getBullet().getIsAvailable() ) {
+		sf::Vector2f pos = player.getSprite().getPosition();
+        player.getBullet().init(1,pos.x+20,pos.y + 15, player.getDirection());
```

Comment:
Make a variable for the offset for where the bullet spawns from, i.e. if player is facing right {xOffset = 20.f} else if player is facing left {xOffset = -20.f}